rotkehlchen/exchanges/cryptocom.py

```diff
@@ -300,133 +350,170 @@ def query_balances(self) -> ExchangeQueryBalances:
-                    f'to query USD price: {e!s}. Skipping balance result.',
-                )
-                continue
-
-            try:
-                available = deserialize_fval(balance_entry['available'])
-                order = deserialize_fval_or_zero(balance_entry.get('order', '0'))
-                stake = deserialize_fval_or_zero(balance_entry.get('stake', '0'))
-                amount = available + order + stake
-            except DeserializationError as e:
-                self.msg_aggregator.add_error(
-                    f'Error processing {self.name} {asset.name} balance result due to inability '
-                    f'to deserialize asset amount due to {e!s}. Skipping balance result.',
-                )
-                continue
+        return dict(assets_balance), ''
 
-            if amount > ZERO:
-                assets_balance[asset] += Balance(
-                    amount=amount,
-                    usd_value=amount * usd_price,
-                )
+    def _query_deposits_withdrawals(
+            self,
+            start_ts: TimestampMS,
+            end_ts: TimestampMS,
+            method: Literal['private/get-deposit-history', 'private/get-withdrawal-history'],
+            query_type: Literal['deposit', 'withdrawal'],
+            success_status: Literal['1', '5'],
+            event_type: Literal[HistoryEventType.DEPOSIT, HistoryEventType.WITHDRAWAL],
+    ) -> list['HistoryBaseEntry']:
+        """Query deposits and withdrawals from the API."""
+        return self._query_paginated(
+            query_type=query_type,
+            method=method,
+            options={
+                'start_ts': start_ts,
+                'end_ts': end_ts,
+                'page_size': DEPOSIT_WITHDRAWAL_PAGE_SIZE,
+                'status': success_status,
+            },
+            initial_page=0,
+            page_key='page',
+            page_size=DEPOSIT_WITHDRAWAL_PAGE_SIZE,
+            result_key=f'{query_type}_list',  # type: ignore[arg-type]  # will be a valid key
+            deserialize_fn=lambda raw_result: self._deserialize_asset_movement(
+                raw_result=raw_result,
+                event_type=event_type,
+            ),
+        )
 
-        return dict(assets_balance), ''
+    def _query_trades(
+            self,
+            start_ts: TimestampMS,
+            end_ts: TimestampMS,
+    ) -> list['HistoryBaseEntry']:
+        """Query trade history from the API.
+        Note that according to the docs, this can only get the history for the last 6 months.
+        https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#introduction-3
+        Also, the time window can only be a max of 1 week per request.
+        https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#private-get-trades
+        """
+        if (now := ts_now_in_ms()) - end_ts > MAX_TRADES_AGE:
+            log.error(f'Cannot query {self.name} trades older than 6 months.')
```

Comment:
This is not an error on our side.
```suggestion
            log.warning(f'Cannot query {self.name} trades older than 6 months.')
```